board/technologic/ts7100/ts7100.c

```diff
@@ -0,0 +1,620 @@
+
+	set_chipselect_size(CS0_128);
+
+	/* Set up I2C bus for uC */
+	setup_i2c(0, CONFIG_SYS_I2C_SPEED, 0x7f, &i2c_pad_info1);
+
+#if !defined(CONFIG_SPL) || defined(CONFIG_SPL_BUILD)
+	setup_iomux_uart();
+#endif
+
+	return 0;
+}
+#endif  /* CONFIG_BOARD_EARLY_INIT_F */
+
+int board_init(void)
+{
+	/* Address of boot parameters */
+	gd->bd->bi_boot_params = PHYS_SDRAM + 0x100;
+#ifdef CONFIG_FEC_MXC
+	reset_phy();
+	setup_fec();
+#endif
+
+        return 0;
+}
+
+int board_late_init(void)
+{
+	char fdtfile[64] = {0};
+	uint32_t cpu_opts;
+	uint32_t io_model;
+	uint32_t io_opts;
+	uint32_t cpu_straps;
+	uint32_t fpga_straps;
+
+        hw_watchdog_reset();
+
+	fram_init();
+
+	imx_iomux_v3_setup_multiple_pads(misc_pads, ARRAY_SIZE(misc_pads));
+
+        /*
+         * WARNING: All of these are wiped out of the environment
+         * after an "env default -a", until the board is reset.
+         */
+	env_set("model", get_board_model());
+
+        fpga_straps = read_raw_fpga_straps();
+	env_set_hex("raw_fpga_straps", fpga_straps);
+
+        cpu_straps = read_raw_cpu_straps();
+	env_set_hex("raw_cpu_straps", cpu_straps);
+
+        cpu_opts = read_cpu_board_opts();
+	env_set_hex("opts", cpu_opts);
+
+	/* Read and parse 6UL pins used for IO board strapping */
+        io_model = read_io_board_model();
+	env_set_hex("io_model", io_model);
+
+	snprintf(fdtfile, sizeof(fdtfile),
+                 "imx6ul-ts%s-%d.dtb", get_board_model(), io_model);
+	env_set("fdtfile", fdtfile);
+
+        io_opts = read_io_board_opts();
+	env_set_hex("io_opts", io_opts);
+
+	env_set("board_name", get_board_name());
+	env_set("board_rev", get_cpu_board_version());
+
+	if(is_mfg()) {
+		env_set("bootcmd", "mfg");
+		env_set("bootdelay", "1");
+	}
+
+	return 0;
+}
+
+#ifdef CONFIG_BOOTCOUNT_LIMIT
+void bootcount_store(ulong a)
+{
+	fram_write(2047, (uint8_t) a);
+	/* Takes 9us to write */
+	udelay(18);
+	if((uint8_t)a != fram_read(2047)) {
+		printf("New FRAM bootcount did not write!\n");
+	}
+}
+
+ulong bootcount_load(void)
+{
+	uint8_t val = fram_read(2047);
+
+	/* Depopulated FRAM returns all ffs */
+	if(val == 0xff) {
+		puts("FRAM returned unexpected value.  Returning bootcount 0\n");
+		return 0;
+	}
+
+	return val;
+}
+
+#endif //CONFIG_BOOTCOUNT_LIMIT
+
+#if defined(CONFIG_FPGA)
+
+static void ts7100_fpga_jtag_init(void)
+{
+	gpio_direction_output(JTAG_FPGA_TDI, 1);
+	gpio_direction_output(JTAG_FPGA_TCK, 1);
+	gpio_direction_output(JTAG_FPGA_TMS, 1);
+	gpio_direction_input(JTAG_FPGA_TDO);
+}
+
+static void ts7100_fpga_done(void)
+{
+	gpio_direction_input(JTAG_FPGA_TDI);
+	gpio_direction_input(JTAG_FPGA_TCK);
+	gpio_direction_input(JTAG_FPGA_TMS);
+	gpio_direction_input(JTAG_FPGA_TDO);
+
+	/* During FPGA programming several important pins will
+	 * have been tristated.  Put it back to normal */
+	//fpga_mmc_init();
+	//red_led_on();
+	//green_led_off();
+	//fpga_gpio_output(OFF_BD_RESET_PADN, 1);
+	//fpga_gpio_output(EN_USB_HOST_5V_PAD, 1);
```

Comment:
I think we can remove these.  If we want to talk about supporting soft/ram reloaded fpgas we should set sane defaults in the fpga itself.